scripts/bot.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import os
+from contextlib import contextmanager
+from functools import cache
+from typing import cast
+
+import anyio
+import gradio as gr
+from agents import (
+    Agent,
+    ModelSettings,
+    OpenAIChatCompletionsModel,
+    Runner,
+    set_default_openai_client,
+    set_default_openai_key,
+    set_tracing_disabled,
+)
+from agents.mcp import MCPServerStdio
+from dotenv import load_dotenv
+from openai import AsyncAzureOpenAI, AsyncOpenAI
+
+
+@cache
+def get_openai_model_settings():
+    temperature = float(os.getenv("OPENAI_TEMPERATURE", 0.0))
+    return ModelSettings(temperature=temperature)
+
+
+@cache
+def get_openai_model() -> OpenAIChatCompletionsModel:
+    model_name = os.getenv("OPENAI_MODEL", "gpt-4o-mini")
+    return OpenAIChatCompletionsModel(
+        model_name,
+        openai_client=get_openai_client(),
+    )
+
+
+@cache
+def get_openai_client() -> AsyncOpenAI:
+    azure_api_key = os.getenv(key="AZURE_OPENAI_API_KEY")
+    if azure_api_key is not None:
+        azure_client = AsyncAzureOpenAI(api_key=azure_api_key)
+        set_default_openai_key(azure_api_key)
+        set_default_openai_client(azure_client)
+
+        # Disable tracing since Azure doesn't support it
+        set_tracing_disabled(True)
+
+        return cast(AsyncOpenAI, azure_client)
+
+    return AsyncOpenAI()
+
+
+class Bot:
+    def __init__(self, mcp_servers: list[MCPServerStdio] | None = None) -> None:
+        self.mcp_servers = mcp_servers or []
+        self.agent = Agent(
+            name="Bot",
+            instructions="使用台灣中文",
```

Comment:
add this to parameter? we may want to use other instructions

Reply:
as your wish